### PR TITLE
feat(ui5-calendar): implement calendar week numbering

### PR DESCRIPTION
**Pull request by tsanislavgatev**
With this feature you can select one of four options to sete to the `Calendar`, `DatePicker`, `DateTimePicker`, `DateRangePicker` components.

Default -Determined by active format locale
ISO_8601 - Official calendar week numbering in most of Europe (ISO 8601standard)
WesternTraditional -Official calendar week numbering in the United States, Canada, Brazil, Israel, Japan, and other countries
MiddleEastern - Official calendar week numbering in much of the Middle East

It can be used as:
```html
<ui5-calendar id="calendar10" calendar-week-numbering="WesternTraditional">
  <ui5-date value="Jan 1, 2023"></ui5-date-range>
</ui5-calendar>
```

This property will control the week numbering and the first day of the week in a way explained in the enumeration documentation.

### CODE REVIEW

File: packages/website/docs/_components_pages/main/Calendar/Calendar.mdx [hunk 1]
```diff
@@ -45,4 +46,8 @@ Discover all the available <Link to={useBaseUrl("/components/CalendarLegend#cale
 
 ### Timezones
 You can set to the configuration the preferred time zone, such as: Asia/Tokyo, Pacific/Apia, Asia/Kolkata, Europe/Sofia and etc.
-<CalendarInDifferentTimezone />
\ No newline at end of file
+<CalendarInDifferentTimezone />
+
+### CalendarWeekNumbering
+You can use the component the preferred week numbering and first day of week.
```
Comment (hinzzx) [inline packages/website/docs/_components_pages/main/Calendar/Calendar.mdx:52]: Very minor one: 
`You can use the component, the preferred week numbering and first day of week.`

File: packages/website/docs/_components_pages/main/DatePicker.mdx [hunk 2]
```diff
@@ -31,4 +32,8 @@ You can set the preferred calendar, such as: Gregorian, Islamic, Japanese and Pe
 
 ### Timezones
 You can set to the configuration the preferred time zone, such as: Asia/Tokyo, Pacific/Apia, Asia/Kolkata, Europe/Sofia and etc.
-<DatePickerInDifferentTimezone />
\ No newline at end of file
+<DatePickerInDifferentTimezone />
+
+### CalendarWeekNumbering
+You can use the component the preferred week numbering and first day of week.
```
Comment (hinzzx) [inline packages/website/docs/_components_pages/main/DatePicker.mdx:38]: same as previous comment (missing comma);

**Review by hinzzx — APPROVED** [review]

File: packages/main/src/types/CalendarWeekNumbering.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/**
+ * The <code>CalendarWeekNumbering</code> enum defines how to calculate calendar weeks. Each
+ * value defines:
+ * <ul>
```
Comment (nnaydenow) [inline packages/main/src/types/CalendarWeekNumbering.ts:4]: use markdown syntax for list

File: packages/main/src/DateComponentBase.ts [hunk 4]
```diff
@@ -81,6 +82,16 @@ class DateComponentBase extends UI5Element {
...
 
+	/**
+	 * Defines how to calculate calendar weeks and first day of the week.
+	 * If not set, the calendar will be displayed according to the currently set global configuration.
+	 * @default "Default"
+	 * @since 2.2.0
+	 * @public
+	 */
+	@property()
+	calendarWeekNumbering?: `${CalendarWeekNumbering}` = "Default";
```
Comment (nnaydenow) [inline packages/main/src/DateComponentBase.ts:93]: ? is not needed when the property has initializer

**Review by nnaydenow — CHANGES_REQUESTED** [review]

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**